backend/docs/migration.py

```diff
@@ -0,0 +1,126 @@
+    else:
+        raise Exception(f"Error fetching pipelines: {response.status_code} - {response.text}")
+
+# Fetch all versions for a given pipeline
+def fetch_pipeline_versions(pipeline_id):
+    url = f"{PIPELINE_API_URL}/{pipeline_id}{PIPELINE_VERSION_API_ENDPOINT}?sort_by=created_at&order_by=asc"
+    response = requests.get(url, headers=HEADERS, verify=VERIFY_CERT)
+    if response.status_code == 200:
+        return response.json()
+    else:
+        raise Exception(f"Error fetching versions for pipeline {pipeline_id}: {response.status_code} - {response.text}")
+
+# Creates a Kubernetes-safe name for a pipeline version by combining the pipeline name and version ID.
+# If the name is too long, it shortens it and adds index at the end to keep it unique.
+def create_version_name(pipeline_name, version_id, index):
+    original_version_name = f"{pipeline_name}-{version_id}"
+    # Converts the name to a valid name
+    formatted_name = utils.maybe_rename_for_k8s(original_version_name)
+
+    if len(formatted_name) <= MAX_NAME_LENGTH:
+        return formatted_name    
+    
+    suffix = f"-{index}"
+    max_pipeline_len = MAX_NAME_LENGTH - len(suffix)
+    trimmed_name = utils.maybe_rename_for_k8s(original_version_name[:max_pipeline_len])
+    return f"{trimmed_name}{suffix}"
+
+# Convert a pipeline and its versions into k8s format
+def convert_to_k8s_format(pipeline, pipeline_versions):
+    k8s_objects = []
+
+    # Converts the name to a valid name
+    pipeline_name = utils.maybe_rename_for_k8s(pipeline.get('display_name', 'unnamed'))
+    versions = pipeline_versions.get("pipeline_versions", [])
+    
+    # Create the Pipeline object
+    pipeline_obj = {
+        "apiVersion": K8S_PIPELINE_API_VERSION,
+        "kind": "Pipeline",
+        "metadata": {
+            "name": pipeline_name,
+            "namespace": K8S_NAMESPACE,
+        }
+    }
+    k8s_objects.append(pipeline_obj)
+
+     # Create a PipelineVersion object for each version
+    for i, version in enumerate(versions):
+        version_id = version.get("pipeline_version_id", f"v{i}")
+        pipeline_version_name = create_version_name(pipeline_name, version_id, i)        
+        pipeline_spec = version.get("pipeline_spec", {})
+        
+        pipeline_version_obj = {
+            "apiVersion": K8S_PIPELINE_API_VERSION,
+            "kind": "PipelineVersion",
+            "metadata": {
+                "name": pipeline_version_name,
+                "namespace": K8S_NAMESPACE
+            },
+            "spec": {
+                "pipelineName": pipeline_name,
+                "pipelineSpec": pipeline_spec
+            }
+        }
+        k8s_objects.append(pipeline_version_obj)
+   
+    return k8s_objects
+
+# Write all collected Kubernetes objects to a YAML file
+def write_yaml_file(k8s_objects, output_path):
+    output_path = Path(output_path)
+    with output_path.open('w') as f:
+        yaml.dump_all(k8s_objects, f, sort_keys=False)
```

Comment:
why do we need to turn off `sort_keys`?

Reply:
To make sure the keys in the output YAML file are not rearranged alphabetically.